### PR TITLE
Blockchain-83 - E-resident checkbox in users identity form

### PR DESCRIPTION
**Pull request by MichalLustykNeti**
image

### CODE REVIEW

File: src/components/Modals/OnchainIdentityModal.js [hunk 1]
```diff
@@ -46,7 +50,10 @@ function OnchainIdentityModal({
```
Comment (kacperzuk-neti) [inline src/components/Modals/OnchainIdentityModal.js:50]: Specify `mode` in `useForm` call: https://react-hook-form.com/docs/useform#mode
I suggest `all`, as that's what we use in other parts of app.

File: src/components/Profile/index.js [hunk 2]
```diff
@@ -51,6 +51,7 @@ function Profile({ className }) {
   };
 
   const handleSubmitOnchainIdentity = (values) => {
+    console.log('values', {values})
```
Comment (kacperzuk-neti) [inline src/components/Profile/index.js:54]: Cleanup.

**Review by kacperzuk-neti — CHANGES_REQUESTED** [review]

File: src/components/Modals/OnchainIdentityModal.js (hunk 1 above)
Comment (kacperzuk-neti) [reply src/components/Modals/OnchainIdentityModal.js:50]: (btw. it's one line above the comment, it's just that GitHub doesn't show enough context.

File: src/components/Modals/OnchainIdentityModal.js [hunk 3]
```diff
@@ -92,27 +96,35 @@ function OnchainIdentityModal({
...
-          label="I am or want to become a e-resident"
-          validate={validateCheckboxs}
-        />
-        </span>
+      <CheckboxInput
+        register={register}
+        name="e_resident"
+        setValue={setValue}
+        watch={watch}
+        trigger={trigger}
```
Comment (kacperzuk-neti) [inline src/components/Modals/OnchainIdentityModal.js:104]: `setValue`, `watch` and `trigger` aren't used by CheckboxInput

File: src/components/Modals/OnchainIdentityModal.js [hunk 4]
```diff
@@ -92,27 +96,35 @@ function OnchainIdentityModal({
...
+         }
+      />
       <div className={styles.title}>Are you or do you want to become a citizen?</div>
-      <span onClick={validateCheckboxs}>
       <CheckboxInput
         register={register}
         name="citizen"
         setValue={setValue}
         watch={watch}
+        trigger={trigger}
```
Comment (kacperzuk-neti) [inline src/components/Modals/OnchainIdentityModal.js:119]: `setValue`, `watch` and `trigger` aren't used by CheckboxInput

**Review by kacperzuk-neti — CHANGES_REQUESTED** [review]

**Review by kacperzuk-neti — APPROVED** [review]

**Review by kacperzuk-neti — CHANGES_REQUESTED** [review]
Rebase with main & squash.

File: src/api/nodeRpcCall.js [hunk 5]
```diff
@@ -164,14 +164,28 @@ const getCitizenAdditionals = (blockNumber, eligible_on_date) => {
...
+  ];
+}
+
+const buildAdditionals = (values, blockNumber) => {
+  const citizenAdditionalsData = values.citizen ? citizenAdditionals(blockNumber, values.eligible_on) : []
+  console.log({citizenAdditionalsData})
+  const eResidentAdditionalsData = values.eresident ? eResidentAdditionals() : []
+  console.log({citizenAdditionalsData})
+
+  console.log('banana', [...citizenAdditionalsData, ...eResidentAdditionalsData])
```
Comment (kacperzuk-neti) [inline src/api/nodeRpcCall.js:179]: Cleanup console logs.

**Review by kacperzuk-neti — APPROVED** [review]

**Review by kacperzuk-neti — APPROVED** [review]

**Review by kacperzuk-neti — APPROVED** [review]

File: src/components/Profile/index.js [hunk 6]
```diff
@@ -205,7 +205,8 @@ function Profile({ className }) {
                   <li>Date of birth: {date_of_birth ?? <em>&lt;empty&gt;</em>}</li>
                 }
                 <li>Citizen: {parseCitizen(info?.additional) ? "YES" : "NO"}</li>
-                <li>Citizenship confirmed: {parseCitizenshipJudgement(judgements) ? "YES" : "NO"}</li>
+                <li>E-resident: {parseEResident(info?.additional) ? "YES" : "NO"}</li>
```
Comment (kacperzuk-neti) [inline src/components/Profile/index.js:208]: `parseEResident` doesn't exist in this PR

**Review by DorianSternVukotic — CHANGES_REQUESTED** [review]
see comments

**Comment by MichalLustykNeti** [comment]
Screen of latest commit https://github.com/liberland/liberland_frontend/pull/95/commits/94b067117e2b2e524b3a4a39cc1b247df131eb84 
image

File: src/components/Modals/OnchainIdentityModal.js [hunk 7]
```diff
@@ -84,24 +90,23 @@ function OnchainIdentityModal({
...
         register={register}
-        name="citizen"
-        setValue={setValue}
-        watch={watch}
-        label="I am or want to become a citizen"
+        name="onChainIdentity"
+        options={[
+          { value: "0", display: "E-resident", selected: !isCitizen && isEResident},
+          { value: "1", display: "Citizen", selected: isCitizen && isEResident},
+          { value: "2", display: "Neither", selected: !isCitizen && !isEResident},
```
Comment (kacperzuk-neti) [inline src/components/Modals/OnchainIdentityModal.js:100]: Can we have more descriptive `value`s here? Like `value: eresident`, `value: citizen` etc.?

Comment (DorianSternVukotic) [reply src/components/Modals/OnchainIdentityModal.js]: Yeah we can have strings here and in the code in general

File: src/components/Modals/OnchainIdentityModal.js [hunk 8]
```diff
@@ -84,24 +93,23 @@ function OnchainIdentityModal({
...
+        name="onChainIdentity"
+        options={[
+          { value: "eresident", display: "E-resident"},
+          { value: "citizen", display: "Citizen"},
+          { value: "neither", display: "Neither"},
+        ]}
       />
 
-      { !isCitizen ? null :
+      { !(onChainIdentity === "1") ? null :
```
Comment (DorianSternVukotic) [inline src/components/Modals/OnchainIdentityModal.js:107]: this wouldnt work with string rename ?

**Review by DorianSternVukotic — CHANGES_REQUESTED** [review]

**Pull request merged**